File: VB/CustomDocument/Custom DocumentManager/CustomDocumentManager.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports DevExpress.XtraBars.Docking2010
Imports DevExpress.XtraBars.Docking2010.Views

Namespace CustomDocumentApplication
	<ToolboxItem(True)> _
	Public Class CustomDocumentManager
		Inherits DocumentManager
		Public Sub New()

		End Sub
		Public Sub New(ByVal container As IContainer)
			MyBase.New(container)

		End Sub

		Protected Overrides Sub RegisterViews()
			MyBase.RegisterViews()
			RegisterView(Of CustomWidgetViewRegistrator)(ViewType.Widget)
		End Sub

		Public Overrides Function CreateView(ByVal type As ViewType) As BaseView
            If type = ViewType.Widget Then
                Return New CustomWidgetView(Container)
            End If
			Return MyBase.CreateView(type)
		End Function
	End Class
End Namespace

File: CS/CustomDocument/Custom DocumentManager/CustomDocumentManager.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using DevExpress.XtraBars.Docking2010;
using DevExpress.XtraBars.Docking2010.Views;

namespace CustomDocumentApplication {
    [ToolboxItem(true)]
    public class CustomDocumentManager : DocumentManager {
        public CustomDocumentManager() {

        }
        public CustomDocumentManager(IContainer container)
            : base(container) {

        }

        protected override void RegisterViews() {
            base.RegisterViews();
            RegisterView<CustomWidgetViewRegistrator>(ViewType.Widget);
        }

        public override BaseView CreateView(ViewType type) {
            if (type == ViewType.Widget) return new CustomWidgetView(Container);
            return base.CreateView(type);
        }
    }
}
